numpy/lib/_stride_tricks_impl.py

```diff
@@ -478,7 +478,7 @@ def broadcast_shapes(*args):
     >>> np.broadcast_shapes((6, 7), (5, 6, 1), (7,), (5, 1, 7))
     (5, 6, 7)
     """
-    arrays = [np.empty(x, dtype=[]) for x in args]
+    arrays = [np.empty(x, dtype=bool) for x in args]
```

Comment:
This may rely on the kernel giving us virtual memory very quickly, but that seems fine.

Reply:
Actually, didn't quite see what the old version did: `dtype=[]` gives a structured dtype with size=0. Why is `dtype=bool` better? Would it be equally fast if we defined the an `empty_dtype=np.dtype([])` outside and used that?

Reply:
Ah, nice catch, that might remove the need for the micro-optimization and give us an (effectively) 0 sized array.
I think it's fine, but if someone follows up, that would be great!

Reply:
See #26599 - the effect is actually quite bad for large array sizes so worth fixing.

Reply:
@mhvk Thanks for picking this up! I agree with the analysis.